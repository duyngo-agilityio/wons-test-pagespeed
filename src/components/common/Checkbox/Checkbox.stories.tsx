import type { Meta, StoryObj } from '@storybook/react';

// themes
import { colors } from '@/themes';

// components
import Checkbox from './index';

const meta: Meta<typeof Checkbox> = {
  title: 'Components/Common/Checkbox',
  component: Checkbox,
  argTypes: {
    color: {
      description: 'Select color of the checkbox',
      options: ['default', 'custom'],
      control: { type: 'radio' },
    },
    afterBgColor: {
      description: 'Background color for checkbox after state',
      control: { type: 'select' },
      options: [colors.blue[500], colors.pink[500], colors.purple[600]],
    },
    size: {
      description: 'Change size of checkbox',
      options: ['md', 'lg'],
      control: { type: 'radio' },
    },
    radius: {
      description: 'Change radius of checkbox',
      options: ['sm', 'md', 'lg'],
      control: { type: 'radio' },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Checkbox>;

export const Default: Story = {
  args: {
    color: 'default',
    size: 'md',
    radius: 'sm',
    afterBgColor: '#1E90FF',
  },
};
