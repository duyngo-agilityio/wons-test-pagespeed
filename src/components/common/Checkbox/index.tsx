'use client';

// Libs
import {
  extendVariants,
  Checkbox as NextUICheckbox,
  CheckboxProps as NextUICheckboxProps,
} from '@nextui-org/react';

export const Checkbox = extendVariants(NextUICheckbox, {
  variants: {
    color: {
      primary: {
        wrapper: [
          'after:bg-500 dark:after:bg-purple-600',
          'after:hover:bg-blue-500 dark:after:hover:bg-purple-600',
          'before:border-blue-800/30 dark:before:border-white/30',
        ],
        icon: 'text-white dark:text-gray-400',
      },
    },
    size: {
      md: {
        label: 'text-sm',
        wrapper: ['w-3.75 h-3.75', 'before:border'],
      },
      lg: {
        label: 'text-base',
        wrapper: ['w-5 h-5', 'before:border'],
      },
    },
    radius: {
      sm: {
        wrapper: ['rounded-sm', 'before:rounded-sm', 'after:rounded-sm'],
      },
    },
  },

  defaultVariants: {
    color: 'primary',
    size: 'md',
    radius: 'sm',
  },
});

export default Checkbox;
