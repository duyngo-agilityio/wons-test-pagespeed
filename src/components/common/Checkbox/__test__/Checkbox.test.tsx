import { render } from '@testing-library/react';

// components
import Checkbox from '../index';

describe('Checkbox component snapshot tests', () => {
  test('renders with default props', () => {
    const { container } = render(<Checkbox />);
    expect(container).toMatchSnapshot();
  });

  test('renders with custom color and afterBgColor', () => {
    const { container } = render(
      <Checkbox color="custom" afterBgColor="#ff6347" />,
    );

    const checkboxElement = container.firstChild;
    expect(checkboxElement).toMatchSnapshot;

    expect(container).toMatchSnapshot();
  });

  test('renders with default color and custom afterBgColor', () => {
    const { container } = render(<Checkbox afterBgColor="#32cd32" />);

    const checkboxElement = container.firstChild;
    expect(checkboxElement).toMatchSnapshot();

    expect(container).toMatchSnapshot();
  });
});
