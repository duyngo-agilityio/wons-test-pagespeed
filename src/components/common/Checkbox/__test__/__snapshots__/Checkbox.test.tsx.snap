// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component snapshot tests renders with custom color and afterBgColor 1`] = `
<div>
  <label
    class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer tap-highlight-transparent p-2 -m-2 custom-checkbox"
    style="--after-bg-default: #ff6347;"
  >
    <span
      style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
    >
      <input
        aria-label=" "
        aria-labelledby=":r1:"
        type="checkbox"
        value=""
      />
    </span>
    <span
      aria-hidden="true"
      class="relative inline-flex items-center justify-center flex-shrink-0 overflow-hidden before:content-[''] before:absolute before:inset-0 before:border-solid before:border-2 before:box-border before:border-default after:content-[''] after:absolute after:inset-0 after:scale-50 after:opacity-0 after:origin-center group-data-[selected=true]:after:scale-100 group-data-[selected=true]:after:opacity-100 group-data-[hover=true]:before:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background after:bg-primary after:text-primary-foreground w-5 h-5 mr-2 rtl:ml-2 rtl:mr-[unset] before:rounded-[calc(theme(borderRadius.medium)*0.5)] after:rounded-[calc(theme(borderRadius.medium)*0.5)] before:transition-colors group-data-[pressed=true]:scale-95 transition-transform after:transition-transform-opacity after:!ease-linear after:!duration-200 motion-reduce:transition-none text-white rounded-sm"
    >
      <svg
        aria-hidden="true"
        class="z-10 opacity-0 group-data-[selected=true]:opacity-100 w-4 h-3 transition-opacity motion-reduce:transition-none"
        role="presentation"
        viewBox="0 0 17 18"
      >
        <polyline
          fill="none"
          points="1 9 7 14 15 4"
          stroke="currentColor"
          stroke-dasharray="22"
          stroke-dashoffset="66"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </span>
  </label>
</div>
`;

exports[`Checkbox component snapshot tests renders with default color and custom afterBgColor 1`] = `
<label
  class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer tap-highlight-transparent p-2 -m-2 custom-checkbox"
  style="--after-bg-default: #32cd32;"
>
  <span
    style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
  >
    <input
      aria-label=" "
      aria-labelledby=":r2:"
      type="checkbox"
      value=""
    />
  </span>
  <span
    aria-hidden="true"
    class="relative inline-flex items-center justify-center flex-shrink-0 overflow-hidden before:content-[''] before:absolute before:inset-0 before:border-solid before:border-2 before:box-border before:border-default after:content-[''] after:absolute after:inset-0 after:scale-50 after:opacity-0 after:origin-center group-data-[selected=true]:after:scale-100 group-data-[selected=true]:after:opacity-100 group-data-[hover=true]:before:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background after:text-default-foreground w-5 h-5 mr-2 rtl:ml-2 rtl:mr-[unset] before:rounded-[calc(theme(borderRadius.medium)*0.5)] after:rounded-[calc(theme(borderRadius.medium)*0.5)] before:transition-colors group-data-[pressed=true]:scale-95 transition-transform after:transition-transform-opacity after:!ease-linear after:!duration-200 motion-reduce:transition-none after:bg-[var(--after-bg-default)] text-white dark:before:border-blue-600 rounded-sm"
  >
    <svg
      aria-hidden="true"
      class="z-10 opacity-0 group-data-[selected=true]:opacity-100 w-4 h-3 transition-opacity motion-reduce:transition-none"
      role="presentation"
      viewBox="0 0 17 18"
    >
      <polyline
        fill="none"
        points="1 9 7 14 15 4"
        stroke="currentColor"
        stroke-dasharray="22"
        stroke-dashoffset="66"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </span>
</label>
`;

exports[`Checkbox component snapshot tests renders with default color and custom afterBgColor 2`] = `
<div>
  <label
    class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer tap-highlight-transparent p-2 -m-2 custom-checkbox"
    style="--after-bg-default: #32cd32;"
  >
    <span
      style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
    >
      <input
        aria-label=" "
        aria-labelledby=":r2:"
        type="checkbox"
        value=""
      />
    </span>
    <span
      aria-hidden="true"
      class="relative inline-flex items-center justify-center flex-shrink-0 overflow-hidden before:content-[''] before:absolute before:inset-0 before:border-solid before:border-2 before:box-border before:border-default after:content-[''] after:absolute after:inset-0 after:scale-50 after:opacity-0 after:origin-center group-data-[selected=true]:after:scale-100 group-data-[selected=true]:after:opacity-100 group-data-[hover=true]:before:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background after:text-default-foreground w-5 h-5 mr-2 rtl:ml-2 rtl:mr-[unset] before:rounded-[calc(theme(borderRadius.medium)*0.5)] after:rounded-[calc(theme(borderRadius.medium)*0.5)] before:transition-colors group-data-[pressed=true]:scale-95 transition-transform after:transition-transform-opacity after:!ease-linear after:!duration-200 motion-reduce:transition-none after:bg-[var(--after-bg-default)] text-white dark:before:border-blue-600 rounded-sm"
    >
      <svg
        aria-hidden="true"
        class="z-10 opacity-0 group-data-[selected=true]:opacity-100 w-4 h-3 transition-opacity motion-reduce:transition-none"
        role="presentation"
        viewBox="0 0 17 18"
      >
        <polyline
          fill="none"
          points="1 9 7 14 15 4"
          stroke="currentColor"
          stroke-dasharray="22"
          stroke-dashoffset="66"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </span>
  </label>
</div>
`;

exports[`Checkbox component snapshot tests renders with default props 1`] = `
<div>
  <label
    class="group relative max-w-fit inline-flex items-center justify-start cursor-pointer tap-highlight-transparent p-2 -m-2 custom-checkbox"
    style="--after-bg-default: #3A36DB;"
  >
    <span
      style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
    >
      <input
        aria-label=" "
        aria-labelledby=":r0:"
        type="checkbox"
        value=""
      />
    </span>
    <span
      aria-hidden="true"
      class="relative inline-flex items-center justify-center flex-shrink-0 overflow-hidden before:content-[''] before:absolute before:inset-0 before:border-solid before:border-2 before:box-border before:border-default after:content-[''] after:absolute after:inset-0 after:scale-50 after:opacity-0 after:origin-center group-data-[selected=true]:after:scale-100 group-data-[selected=true]:after:opacity-100 group-data-[hover=true]:before:bg-default-100 outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-focus group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-offset-background after:text-default-foreground w-5 h-5 mr-2 rtl:ml-2 rtl:mr-[unset] before:rounded-[calc(theme(borderRadius.medium)*0.5)] after:rounded-[calc(theme(borderRadius.medium)*0.5)] before:transition-colors group-data-[pressed=true]:scale-95 transition-transform after:transition-transform-opacity after:!ease-linear after:!duration-200 motion-reduce:transition-none after:bg-[var(--after-bg-default)] text-white dark:before:border-blue-600 rounded-sm"
    >
      <svg
        aria-hidden="true"
        class="z-10 opacity-0 group-data-[selected=true]:opacity-100 w-4 h-3 transition-opacity motion-reduce:transition-none"
        role="presentation"
        viewBox="0 0 17 18"
      >
        <polyline
          fill="none"
          points="1 9 7 14 15 4"
          stroke="currentColor"
          stroke-dasharray="22"
          stroke-dashoffset="66"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </span>
  </label>
</div>
`;
